From 058f7035e05927cdf1de08e893993d2220eec679 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Fri, 25 Mar 2022 11:17:39 -0300
Subject: [PATCH 139/163] elf: Suppress clang warning on tst-unique3

clang warns that the instatiation of the variable is required,
but no definition is available.  They are implemented on
tst-unique3lib.so.

Checked on x86_64-linux-gnu.
---
 elf/tst-unique3.cc | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/elf/tst-unique3.cc b/elf/tst-unique3.cc
index efdd6d78c2..d07601fe6d 100644
--- a/elf/tst-unique3.cc
+++ b/elf/tst-unique3.cc
@@ -1,9 +1,14 @@
-#include "tst-unique3.h"
-
 #include <cstdio>
-#include "../dlfcn/dlfcn.h"
+#include <dlfcn.h>
+#include <libc-diag.h>
+#include "tst-unique3.h"
 
+/* clang warns that the instatiation of the variable is required, but no
+   definition is available.  They are implemented on tst-unique3lib.so.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wundefined-var-template");
 int t = S<char>::i;
+DIAG_POP_NEEDS_COMMENT_CLANG;
 
 int
 main (void)
-- 
2.37.2

